src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -166,6 +166,19 @@ public async Task<IActionResult> Success(string orderId)
         return View(order);
     }
 
+    [AllowAnonymous]
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public async Task<IActionResult> CheckoutWithoutPayment()
+    {
+        var order = await _stripePaymentService.CreateNoPaymentOrderFromShoppingCartAsync(_updateModelAccessor);
```

Comment:
The `CreateNoPaymentOrderFromShoppingCartAsync` can return null too, and then this will cause an NRE.
You should do a null check and return `NotFound`.